synapse/rest/media/v1/oembed.py

```diff
@@ -180,6 +182,10 @@ def parse_oembed_response(self, url: str, raw_body: bytes) -> OEmbedResult:
             elif oembed_type == "video":
                 open_graph_response["og:type"] = "video.other"
                 calc_description_and_urls(open_graph_response, oembed["html"])
+                # Some websites like tenor.com do not respond with the video but
+                # with an iframe, which we don't support.
+                if not open_graph_response["og:video"]:
+                    return None
```

Comment:
I think this might break YouTube previews from using oEmbed? I'd be curious how those look before/after this change (vaguely related to #9733).

I wonder if in this case it would make sense to merge the results of the oEmbed with what we can crawl out of the page?

Reply:
YouTube preview actually work better, before:
![image](https://user-images.githubusercontent.com/1370530/148262370-6a3df2c6-f71c-448a-92d6-7ab19cd292b2.png)
after:
![image](https://user-images.githubusercontent.com/1370530/148262408-9591c62c-e687-4676-884f-f02847383ec6.png)

> I wonder if in this case it would make sense to merge the results of the oEmbed with what we can crawl out of the page?

I think that's a good idea! I'll try to rewrite my patch.